src/main/java/org/opensearch/neuralsearch/plugin/query/NeuralQueryBuilder.java

```diff
@@ -172,10 +193,32 @@ private static void parseQueryParams(XContentParser parser, NeuralQueryBuilder n
         }
     }
 
+    @Override
+    protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) {
+        if (vectorSupplier() != null && vectorSupplier.get() != null) {
+            return new KNNQueryBuilder(fieldName(), vectorSupplier.get(), k());
+        }
```

Comment:
why do we need these checks, if the query builder is running the vectorSupplier will be null isn't it?

Reply:
VectorSupplier is null initially and then gets set during rewrite. The actual vector will not get set until the async call finishes completely.